eth/tracers/api.go

```diff
@@ -799,10 +799,19 @@ func containsTx(block *types.Block, hash common.Hash) bool {
 // TraceTransaction returns the structured logs created during the execution of EVM
 // and returns them as a JSON object.
 func (api *API) TraceTransaction(ctx context.Context, hash common.Hash, config *TraceConfig) (interface{}, error) {
-	_, blockHash, blockNumber, index, err := api.backend.GetTransaction(ctx, hash)
+	// GetTransaction returns 0 for the blocknumber if the transaction is not found
```

Comment:
You mean `tx == nil`? Returning a `nil` result without error when the key could not be found is a quite common pattern in the raw db reads in geth.

Reply:
Just seemed odd to me that there's an error in the function signatures return types, but the return value is always nil.